### PR TITLE
LPS-140233 Prevent some Entry get a RelatedEntry from another one, when OneToMany Relationship

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPS-140233

Allow filter Entries through both full name or shortName of relationship field(OnetoMany).(API)
Fix predicate of unrelatedEntries filter(UI)

### CODE REVIEW

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/entity/v1_0/ObjectEntryEntityModel.java [hunk 1]
```diff
@@ -109,6 +115,17 @@ public ObjectEntryEntityModel(List<ObjectField> objectFields) {
...
+						objectField.getRelationshipType(), "manyToMany")) {
+
+				String objectFieldName = objectField.getName();
+
+				_entityFieldsMap.put(
+					objectFieldName,
+					new IdEntityField(
+						objectFieldName, locale -> objectFieldName,
+						String::valueOf));
+			}
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/entity/v1_0/ObjectEntryEntityModel.java:128]: Hi @p-albuquerque,

this conditional I believe we can remove.

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:stop

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
I'm closing this issue because it will be fixed in this PR: https://github.com/liferay-objects/liferay-portal/pull/1006

**Pull request closed without merging**